generators/csharp/sdk/src/endpoint/snippets/EndpointSnippetsGenerator.ts

```diff
@@ -0,0 +1,111 @@
+    private readonly httpEndpointGenerator: HttpEndpointGenerator;
+    private readonly snippets: Map<string, EndpointSnippets>;
+    constructor({ context }: { context: SdkGeneratorContext }) {
+        this.context = context;
+        this.grpcEndpointGenerator = new GrpcEndpointGenerator({ context });
+        this.httpEndpointGenerator = new HttpEndpointGenerator({ context });
+        this.snippets = this.generate();
+    }
+
+    public getSnippetsForEndpoint(endpointId: string): EndpointSnippets | undefined {
+        return this.snippets.get(endpointId);
+    }
+
+    private generate(): Map<string, EndpointSnippets> {
+        const endpointSnippetsById = new Map<string, EndpointSnippets>();
+        for (const [serviceId, service] of Object.entries(this.context.ir.services)) {
+            const grpcClientInfo =
+                serviceId != null ? this.context.getGrpcClientInfoForServiceId(serviceId) : undefined;
+            for (const endpoint of service.endpoints) {
+                const autogenerated = endpoint.autogeneratedExamples
+                    .map((example) =>
+                        this.generateSingleEndpointSnippet({
+                            endpoint,
+                            isGrpc: grpcClientInfo != null,
+                            example: example.example,
+                            serviceId
+                        })
+                    )
+                    .filter((snippet): snippet is SingleEndpointSnippet => snippet != null);
+                const userSpecified = endpoint.userSpecifiedExamples
+                    .map((example) =>
+                        example.example != null
+                            ? this.generateSingleEndpointSnippet({
+                                  endpoint,
+                                  isGrpc: grpcClientInfo != null,
+                                  example: example.example,
+                                  serviceId
+                              })
+                            : undefined
+                    )
+                    .filter((snippet): snippet is SingleEndpointSnippet => snippet != null);
+                if (autogenerated.length > 0 || userSpecified.length > 0) {
+                    endpointSnippetsById.set(endpoint.id, { autogenerated, userSpecified });
+                }
+            }
+        }
+        return endpointSnippetsById;
+    }
+
+    public generateSingleEndpointSnippet({
+        endpoint,
+        isGrpc,
+        example,
+        serviceId
+    }: {
+        endpoint: HttpEndpoint;
+        isGrpc: boolean;
+        example: ExampleEndpointCall;
+        serviceId: string;
+    }): SingleEndpointSnippet | undefined {
+        const methodInvocation = isGrpc
+            ? this.grpcEndpointGenerator.generateGrpcEndpointSnippet({
+                  example,
+                  endpoint,
+                  serviceId,
+                  clientVariableName: "client",
```

Comment:
nit: Should this be some `const` somewhere? If it's used elsewhere in the generator, I'd imagine it could be part of the `context`, or if not just as an un-exported `const` at the top of this file.

Reply:
yeah agree there's some weirdness here. when i wrote the method, i wasn't sure if this could vary by context and i'd want to be able to pass in the name of the client variable. turns out i've just always used "client" 

i'll move this to the context

Reply:
actually, in the mock server tests, i pass in `Client` because it's a property there. going to leave this as is unless you have other suggestions

Reply:
Works for me, thanks for taking a closer look!